backend/tests/services/test_local_device_command_service.py

```
@@ -356,48 +356,103 @@ def test_local_device_command_registry_default_includes_diagnostic_commands():
+        encoding="utf-8",
+    )
+    return root
+
+
+def test_codex_threads_list_command_reads_sdk_thread_list(tmp_path):
+    """codex_threads_list should expose SDK thread summaries without transcripts."""
+    from app.services.device.command_registry import CODEX_THREADS_LIST_SCRIPT
+
+    fake_sdk_root = _write_fake_openai_codex_sdk(tmp_path / "fake-sdk")
 
     result = subprocess.run(
```

Ruff (0.15.17)

[error] `subprocess` call: check for execution of untrusted input

(S603)